### PR TITLE
DMP-3670: Enable not-null constraints on security_group table

### PR DESCRIPTION
**Pull request by sabahirfan**
### JIRA link (if applicable) ###

DMP-3670

### Change description ###
Enable not-null constraints on security_group table

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/integrationTest/java/uk/gov/hmcts/darts/authorisation/controller/AuthorisationControllerIntTest.java [hunk 1]
```diff
@@ -36,6 +38,7 @@
 import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
 
 @AutoConfigureMockMvc
+@Transactional
```
Comment (cmagowan-hmcts) [inline src/integrationTest/java/uk/gov/hmcts/darts/authorisation/controller/AuthorisationControllerIntTest.java:41]: we have been trying to remove @Transactional from all our tests, so try not to add any.

File: src/integrationTest/java/uk/gov/hmcts/darts/usermanagement/controller/PatchSecurityGroupIntTest.java [hunk 2]
```diff
@@ -30,6 +31,7 @@
 import static uk.gov.hmcts.darts.test.common.data.UserAccountTestData.minimalUserAccount;
 
 @AutoConfigureMockMvc
+@Transactional
```
Comment (cmagowan-hmcts) [inline src/integrationTest/java/uk/gov/hmcts/darts/usermanagement/controller/PatchSecurityGroupIntTest.java:34]: Please don't add @Tranasactional to tests

File: src/integrationTest/java/uk/gov/hmcts/darts/usermanagement/controller/PostSecurityGroupIntTest.java [hunk 3]
```diff
@@ -30,6 +31,7 @@
 import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
 
 @AutoConfigureMockMvc
+@Transactional
```
Comment (cmagowan-hmcts) [inline src/integrationTest/java/uk/gov/hmcts/darts/usermanagement/controller/PostSecurityGroupIntTest.java:34]: Please don't add @Tranasactional to tests

File: src/integrationTest/java/uk/gov/hmcts/darts/authorisation/controller/AuthorisationControllerIntTest.java (hunk 1 above)
Comment (sabahirfan) [reply src/integrationTest/java/uk/gov/hmcts/darts/authorisation/controller/AuthorisationControllerIntTest.java:41]: When you run these tests individually without the @Transactional annotation, they pass successfully. However, when you execute the tests together as a suite, you encounter failures related to detached entity errors.

**Review by cmagowan-hmcts — APPROVED** [review]

**Pull request merged**